urla134d66615_fikr4n_itl_java_tgz-pJ8-example_ExampleJ8/src/org/arabeyes/itl/prayer/astro/AstroLib.java

```diff
@@ -0,0 +1,491 @@
+		
+		if (date.getMonthCompat() <= 2) {
+			jdY--;
+			jdM += 12;
+		}
+		
+		if (date.getYear() < 1)
+			jdY++;
+		
+		if ((date.getYear() > 1582)
+				|| ((date.getYear() == 1582) && ((date.getMonthCompat() > 10) || ((date
+						.getMonthCompat() == 10) && (date.getDay() >= 4)))))
+			jdB = 2 - Math.floor(jdY / 100.0) + Math.floor((jdY / 100.0) / 4.0);
+		
+		JD = Math.floor(365.25 * (jdY + 4716.0))
+		+ Math.floor(30.6001 * (jdM + 1)) + (date.getDay() + (-gmt) / 24.0)
+		+ jdB - 1524.5;
+		
+		return JD;
+	}
+	
+	public static void getAstroValuesByDay(double julianDay,
+			final Location loc, Astro astro, Astro topAstro) {
+		AstroDay ad = new AstroDay();
+		
+		if (astro.getJd() == julianDay - 1) {
+			astro.getRa()[0] = astro.getRa()[1];
+			
+			astro.getRa()[1] = astro.getRa()[2];
+			astro.getDec()[0] = astro.getDec()[1];
+			astro.getDec()[1] = astro.getDec()[2];
+			astro.getSid()[0] = astro.getSid()[1];
+			astro.getSid()[1] = astro.getSid()[2];
+			astro.getDra()[0] = astro.getDra()[1];
+			astro.getDra()[1] = astro.getDra()[2];
+			astro.getRsum()[0] = astro.getRsum()[1];
+			astro.getRsum()[1] = astro.getRsum()[2];
+			computeAstroDay(julianDay + 1, ad);
+			astro.getRa()[2] = ad.getRa();
+			astro.getDec()[2] = ad.getDec();
+			astro.getSid()[2] = ad.getSidtime();
+			astro.getDra()[2] = ad.getDra();
+			astro.getRsum()[2] = ad.getRsum();
+		} else if (astro.getJd() == julianDay + 1) {
+			
+			astro.getRa()[2] = astro.getRa()[1];
+			astro.getRa()[1] = astro.getRa()[0];
+			astro.getDec()[2] = astro.getDec()[1];
+			astro.getDec()[1] = astro.getDec()[0];
+			astro.getSid()[2] = astro.getSid()[1];
+			astro.getSid()[1] = astro.getSid()[0];
+			astro.getDra()[2] = astro.getDra()[1];
+			astro.getDra()[1] = astro.getDra()[0];
+			astro.getRsum()[2] = astro.getRsum()[1];
+			astro.getRsum()[1] = astro.getRsum()[0];
+			computeAstroDay(julianDay - 1, ad);
+			astro.getRa()[0] = ad.getRa();
+			astro.getDec()[0] = ad.getDec();
+			astro.getSid()[0] = ad.getSidtime();
+			astro.getDra()[0] = ad.getDra();
+			astro.getRsum()[0] = ad.getRsum();
+		} else if (astro.getJd() != julianDay) {
+			computeAstroDay(julianDay - 1, ad);
+			astro.getRa()[0] = ad.getRa();
+			astro.getDec()[0] = ad.getDec();
+			astro.getSid()[0] = ad.getSidtime();
+			astro.getDra()[0] = ad.getDra();
+			astro.getRsum()[0] = ad.getRsum();
+			computeAstroDay(julianDay, ad);
+			astro.getRa()[1] = ad.getRa();
+			astro.getDec()[1] = ad.getDec();
+			astro.getSid()[1] = ad.getSidtime();
+			astro.getDra()[1] = ad.getDra();
+			astro.getRsum()[1] = ad.getRsum();
+			computeAstroDay(julianDay + 1, ad);
+			astro.getRa()[2] = ad.getRa();
+			astro.getDec()[2] = ad.getDec();
+			astro.getSid()[2] = ad.getSidtime();
+			astro.getDra()[2] = ad.getDra();
+			astro.getRsum()[2] = ad.getRsum();
+		}
+		
+		astro.setJd(julianDay);
+		computeTopAstro(loc, astro, topAstro);
+		
+	}
+	
+	public static void computeAstroDay(double JD, AstroDay astroday) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 89 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 323-399 into a separate method.